coalib/misc/CachingUtilities.py

```diff
@@ -0,0 +1,176 @@
+import hashlib
+import os
+import pickle
+
+from coalib.misc import Constants
+
+
+def get_data_path(log_printer, identifier):
+    """
+    Get the full path of ``identifier`` present in the user's cache directory.
+
+    :param log_printer: A LogPrinter object to use for logging.
+    :param identifier:  The file whose path needs to be expanded.
+    :return:            Full path of the file, assuming it's present in the
+                        user's config directory.
+                        Returns ``None`` if there is a ``PermissionError``
+                        in creating the directory.
+    """
+    try:
+        os.makedirs(Constants.USER_DATA_DIR, exist_ok=True)
+        return os.path.join(Constants.USER_DATA_DIR, hash_id(identifier))
+    except PermissionError:
+        log_printer.err("Unable to create user data directory '{}'. Continuing"
+                        " without caching.".format(Constants.USER_DATA_DIR))
+
+    return None
+
+
+def delete_cache_files(log_printer, identifiers):
+    """
+    Delete the cache files after displaying a warning saying the cache
+    is corrupted and will be removed.
```

Comment:
this function has nothing to do in particular with corrupted cache, right? I wouldn't include that in the doc comment because it's not specific to that.